pkg/controller/data_download_controller.go

```diff
@@ -289,18 +289,17 @@ func (r *DataDownloadReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		if dd.Spec.Cancel {
 			fsRestore := r.dataPathMgr.GetAsyncBR(dd.Name)
 			if fsRestore == nil {
+				r.OnDataDownloadCancelled(ctx, dd.GetNamespace(), dd.GetName())
 				return ctrl.Result{}, nil
 			}
-
 			log.Info("Data download is being canceled")
-			// Update status to Canceling.
+			// Update status to Canceled.
 			original := dd.DeepCopy()
-			dd.Status.Phase = velerov2alpha1api.DataDownloadPhaseCanceling
+			dd.Status.Phase = velerov2alpha1api.DataDownloadPhaseCanceled
```

Comment:
Before `fsRestore` cancels itself, we should not set DD to `DataDownloadPhaseCanceled`. We need to wait `fsRestore` to cancel itself and then it calls `OnDataDownloadCancelled`. 

So just rollback to the original code for this case.

Reply:
Done, Updated the PR.